### PR TITLE
fix: enhance GA account

### PR DESCRIPTION
**Pull request by azmimuhammad**
### JIRA LINK 🟦
- https://blocksphere2020.atlassian.net/browse/DBIO-2424

### Changelog/Description 🗒️
- fix add/edit certificate & enhance function name from polkadot-povider lib

### WARNING ⚠️
-

### CODE REVIEW

File: src/views/Dashboard/GeneticAnalyst/Account/index.vue [hunk 1]
```diff
@@ -695,36 +697,47 @@ export default {
     async onSubmitFile() {
       this._touchForms("document")
 
-      const { title, issuer, month, year, description, supporting_document } = this.document
+      const { title, issuer, month, year, description, file, supportingDocument } = this.document
 
-      if (!title || !issuer || !month || !year || !supporting_document) return this.errorDoc = true
+      if (!title || !issuer || !month || !year) return this.errorDoc = true
```
Comment (bukandicki) [inline src/views/Dashboard/GeneticAnalyst/Account/index.vue:702]: Remove `return` and please update to `if (!title || !issuer || !month || !year) this.errorDoc = true`

**Review by bukandicki — CHANGES_REQUESTED** [review]

File: src/views/Dashboard/GeneticAnalyst/Account/index.vue (hunk 1 above)
Comment (azmimuhammad) [reply src/views/Dashboard/GeneticAnalyst/Account/index.vue:702]: if remove the `return` code after it will be executed, WDYT @bukandicki

**Review by agustinustheo — APPROVED** [review]

File: src/views/Dashboard/GeneticAnalyst/Account/index.vue (hunk 1 above)
Comment (bukandicki) [reply src/views/Dashboard/GeneticAnalyst/Account/index.vue:702]: then you need to declare it after your `this.errorDoc` like

```javascript
if (!title || !issuer || !month || !year) {
  this.errorDoc = true
  return
}
```

please don't return setter, use return as a value or to prevent something, this is a bad way, so you need to update, even the sonarcloud say it as code smells

Comment (azmimuhammad) [reply src/views/Dashboard/GeneticAnalyst/Account/index.vue]: updated @bukandicki , thanks for the advice

File: src/common/lib/pinata-proxy.js [hunk 2]
```diff
@@ -17,31 +15,15 @@ export const uploadFile = val => {
...
-    pinataJwtKey,
-    source.token,
-    (progressEvent) => {
-      if(progressEvent.lengthComputable) {
-        let percentCompleted = Math.round( (progressEvent.loaded * 100) / progressEvent.total )
-        store.dispatch("geneticData/getLoadingProgress", {
-          progress: percentCompleted
-        })
-      }
-    }
```
Comment (bukandicki) [inline src/common/lib/pinata-proxy.js:39]: just curious, why did you delete these codes? and have you checked the whole environment does not work properly if these codes are removed?

**Review by bukandicki — CHANGES_REQUESTED** [review]

File: src/common/lib/pinata-proxy.js (hunk 2 above)
Comment (azmimuhammad) [reply src/common/lib/pinata-proxy.js:39]: because previously there was an error in cancelToken function, and after I checked this function was just used on this page

Comment (bukandicki) [reply src/common/lib/pinata-proxy.js:39]: what kind of error? can you mention the error here? because I doubt it will run properly on other pages

Comment (azmimuhammad) [reply src/common/lib/pinata-proxy.js:39]: oh Sorry, after I checked again, it turned out that there was no error. i will revert the code

Comment (azmimuhammad) [reply src/common/lib/pinata-proxy.js]: updated @bukandicki

File: src/common/lib/pinata-proxy.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ export const uploadFile = val => {
 }
 
 export const getFileUrl = cid => {
-  return `${process.env.VUE_APP_PINATA_GATEWAY}/${cid}`
+  return `${process.env.VUE_APP_PINATA_GATEWAY}/ipfs/${cid}`
```
Comment (bukandicki) [inline src/common/lib/pinata-proxy.js:44]: should we use `/ipfs`? I think you need to see this code in the customer repository, I don't think there is a difference there, then you should reference from there

**Review by bukandicki — CHANGES_REQUESTED** [review]

File: src/common/lib/pinata-proxy.js (hunk 3 above)
Comment (azmimuhammad) [reply src/common/lib/pinata-proxy.js:44]: if not use `/ipfs` url is ivalid
image


if use `/ipfs` 
image

Comment (bukandicki) [reply src/common/lib/pinata-proxy.js:44]: which ENV are you using?

Comment (azmimuhammad) [reply src/common/lib/pinata-proxy.js]: oo i see
